python/pyspark/sql/column.py

```diff
@@ -405,6 +415,20 @@ def getItem(self, key: Any) -> "Column":
 
         .. versionadded:: 1.3.0
 
+        Parameters
+        ----------
+        key
+            a literal value, or a :class:`Column` expression.
+            The result will only be true at a location if item matches in the column.
+
+             .. deprecated:: 3.0.0
+                 :class:`Column` as a parameter is deprecated.
+
+        Returns
+        -------
+        :class:`Column`
+            Column representing whether each element of Column as items contained in key.
```

Comment:
nit: how about `Column representing the item(s) got at position ``ordinal`` out of a list or by key out of a dict`?